server/s3_object_metadata.cc

```diff
@@ -364,20 +414,35 @@ void S3ObjectMetadata::load_successful() {
 }
 
 void S3ObjectMetadata::load_failed() {
-  if (json_parsing_error) {
-    state = S3ObjectMetadataState::failed;
-  } else if (motr_kv_reader->get_state() == S3MotrKVSReaderOpState::missing) {
-    s3_log(S3_LOG_DEBUG, request_id, "Object metadata missing for %s\n",
-           object_name.c_str());
-    state = S3ObjectMetadataState::missing;  // Missing
-  } else if (motr_kv_reader->get_state() ==
-             S3MotrKVSReaderOpState::failed_to_launch) {
-    s3_log(S3_LOG_WARN, request_id, "Object metadata load failed\n");
-    state = S3ObjectMetadataState::failed_to_launch;
-  } else {
-    s3_log(S3_LOG_WARN, request_id, "Object metadata load failed\n");
-    state = S3ObjectMetadataState::failed;
+  switch (motr_kv_reader->get_state()) {
+    case S3MotrKVSReaderOpState::failed_to_launch:
+      state = S3ObjectMetadataState::failed_to_launch;
+      s3_log(S3_LOG_WARN, request_id,
+             "Object metadata load failed to launch - ServiceUnavailable");
```

Comment:
Do ensure that in latest logging (Anatoliy's changes) \n is put in at end

Reply:
fixed